src/components/base/BaseButton.vue

```diff
@@ -3,39 +3,34 @@ import { computed } from "vue";
 
 const props = withDefaults(
   defineProps<{
-    buttonType:
-    | "primary"
-    | "primary-rev"
-    | "secondary"
-    | "tertiary"
-    | "tertiary-rev"
-    | "breadcrumb"
-    | "breadcrumb-rev";
-    buttonSize: "small" | "regular" | "medium" | "large" | "slim";
-    to: string;
+    buttonKind?: "button" | "submit";
+    buttonType?: "primary" | "secondary" | "tertiary" | "breadcrumb";
+    buttonSize?: "small" | "regular" | "medium" | "large" | "slim";
+    buttonStyle?: "outlined" | "noborder" | "";
+    to?:
```

Comment:
Actually Vue Router has this type defined and we can import it here. Try to import this one:
https://router.vuejs.org/api/index.html#routelocationraw

Reply:
I would mind advise on this if you have a minute. I defined route to as 'RouteLocationRaw' as per your suggestion but i am facing crop of warnings
`[Vue warn]: Invalid prop: type check failed for prop "to". Expected String | Null, got Object `

Reply:
Hmmm it should work out of the box 🤔
Could you give an example on how you are trying to use it and where?